session/database.rst

```diff
@@ -0,0 +1,600 @@
+working MongoDB connection in your Symfony application as explained in the
+`DoctrineMongoDBBundle configuration`_ article.
+
+Then, register a new handler service for ``MongoDbSessionHandler`` and pass it
+the MongoDB connection as argument:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+
+            Symfony\Component\HttpFoundation\Session\Storage\Handler\MongoDbSessionHandler:
+                arguments:
+                    - '@doctrine_mongodb.odm.default_connection'
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <services>
+                <service id="Symfony\Component\HttpFoundation\Session\Storage\Handler\MongoDbSessionHandler" public="false">
+                    <argument type="service">doctrine_mongodb.odm.default_connection</argument>
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\HttpFoundation\Session\Storage\Handler\MongoDbSessionHandler;
+
+        $storageDefinition = $container->autowire(MongoDbSessionHandler::class)
+            ->setArguments([
+                new Reference('doctrine_mongodb.odm.default_connection'),
+            ])
+        ;
+
+Next, use the :ref:`handler_id <config-framework-session-handler-id>`
+configuration option to tell Symfony to use this service as the session handler:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            session:
+                # ...
+                handler_id: Symfony\Component\HttpFoundation\Session\Storage\Handler\MongoDbSessionHandler
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
+        <framework:config>
+            <!-- ... -->
+            <framework:session
+                handler-id="Symfony\Component\HttpFoundation\Session\Storage\Handler\MongoDbSessionHandler"/>
+        </framework:config>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        use Symfony\Component\HttpFoundation\Session\Storage\Handler\MongoDbSessionHandler;
+
+        // ...
+        $container->loadFromExtension('framework', [
+            // ...
+            'session' => [
+                'handler_id' => MongoDbSessionHandler::class,
+            ],
+        ]);
+
+.. note::
+
+    MongoDB ODM 1.x only works with the legacy driver, which is no longer
```

Comment:
Why taking about MongoDB ODM here ? This handler is not using the ODM at all.

Reply:
The service uses the MongoDB connection service created by ODM, which is why it's worth mentioning. The alternative is creating a separate `MongoDB\Client` instance for use with sessions, which users may do. Maybe that's worth explaining as well?

Reply:
> The service uses the MongoDB connection service created by ODM, which is why it's worth mentioning.

I don't understand this statement. The MongodbSessionHandler expects a `\MongoDB\Client` or `\MongoClient` or `\Mongo`. It does not use the Doctrine wrappers.

Reply:
> I don't understand this statement. The MongodbSessionHandler expects a `\MongoDB\Client` or `\MongoClient` or `\Mongo`.

The MongoDB session handler expects a `\MongoDB\Client` object since Symfony 4 - using the `\MongoClient` or the deprecated `\Mongo` class was removed in 4.0 as those classes aren't readily available on PHP 7 (as the legacy driver was never made compatible with PHP 7).

> It does not use the Doctrine wrappers.

That is correct, but the service configuration shown in the docs uses the default connection that is created when using MongoDB ODM. This reuses connection info like the URL, auth details, and more to avoid users having to duplicate that info. Of course, users may very well set up a separate connection for this purpose in ODM or create their own `\MongoDB\Client` instance to configure the service.

One could go and set up the MongoDB session storage without ODM, but chances are people using MongoDB for session storage might also be using it for something else, in which case the ODM is the recommended way to go.

@javiereguiluz do you think there should be instructions for setting up the client without using the ODM? While I think it could be helpful for some people, I'm worried that it will confuse people as to what version to choose. On the other hand, for people that aren't using ODM this might give the impression that they need the ODM solely for the purpose of storing sessions, which is not correct.

Reply:
I'm afraid I don't understand what you are discussing about. Please, tell me what should I do here: remove this note? Reword it? Thanks.